### PR TITLE
ISSUE-13: [refactor] stop replication by close repl channel

### LINKED ISSUE #13: Notify replication task to stop by just closing the channel.
`RaftEvent::Terminate` is only used to notify replication to stop, when Leader switches to Follower.

Remove this `RaftEvent::Terminate`. To stop replication, just closing the channel is ok.

https://github.com/datafuse-extras/async-raft/blob/9adb5ff5d9cd27366545c1d0040e979239287c5f/async-raft/src/core/mod.rs#L727-L730

### PR DESCRIPTION
**Pull request by ariesdevil**
**Summary**
when switching from leader to follower, just close the channel instead of sending `RaftEvent::Terminate`.

**Related issues**
Fix #13 

PS: I found toolchain with `nightly-2021-12-30` will cause rustc panic randomly, so I update the toolchain

### CODE REVIEW

File: openraft/src/core/mod.rs [hunk 1]
```diff
@@ -739,10 +738,9 @@ impl<'a, D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>
         loop {
             if !self.core.target_state.is_leader() {
                 tracing::info!("id={} state becomes: {:?}", self.core.id, self.core.target_state);
+                // notify to all nodes DO NOT send replication event any more.
+                self.replication_rx.close();
```
Comment (drmingdrmer) [inline openraft/src/core/mod.rs:742]: It seems like a single `return` would be enough: channel will be closed when dropped.

Although I am not sure if it's a good idea to do that 🤔 
What say you man?

Comment (ariesdevil) [reply openraft/src/core/mod.rs:742]: The difference between `drop` and `close` is that a closed receiver can still drain messages that the sender already ended to the channel buffer. In this case, `drop` is enough.
So you are totally right as always.

**Comment by drmingdrmer** [comment]
> The difference between `drop` and `close` is that a closed receiver can still drain messages that the sender already ended to the channel buffer. In this case, `drop` is enough. So you are totally right as always.

@ariesdevil 
I did not know that. Lesson learned. 🤔

**Review by drmingdrmer — APPROVED** [review]
Less codes better!

**Pull request merged**